File: template-pattern/python/src/Basketball.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""
from src.GameTemplate import GameTemplate


# 定义子类覆写父类抽象方法
class Basketball(GameTemplate):

    def init(self):
        print(
            'Basketball::init() [Basketball Game Initialized! Start playing.]')

    def start(self):
        print('Basketball::start() [Basketball Game Started. Enjoy the game!]')

    def end(self):
        print('Basketball::end() [Basketball Game Finished!]')

File: template-pattern/go/src/Basketball.go
package src

import "fmt"

// 定义子类覆写父类抽象方法
type Basketball struct {
  // 继承父模板的公共方法
  GameTemplate
}

func (b *Basketball) GetName() string {
  return "Basketball"
}

func (b *Basketball) Init() {
  fmt.Println("Basketball::Init() [Basketball Game Initialized! Start playing.]")
}

func (b *Basketball) Start() {
  fmt.Println("Basketball::Start() [Basketball Game Started. Enjoy the game!]")
}

func (b *Basketball) End() {
  fmt.Println("Basketball::End() [Basketball Game Finished!]")
}
